resolwe/flow/views.py

```diff
@@ -214,6 +216,35 @@ def set_public_permissions(perm_type):
         set_public_permissions('add')
         set_public_permissions('remove')
 
+    def _get_object_permissions(self, user, instance):
+        def format_permissions(perms):
+            return [perm.replace('_{}'.format(instance._meta.model_name), '') for perm in perms]
+
+        if user.is_authenticated():
+            permissions_user, permissions_group = get_user_group_perms(user, instance)
+        else:
+            permissions_user, permissions_group = [], []
+        permissions_public = shortcuts.get_perms(AnonymousUser(), instance)
+
+        return {
+            'user': format_permissions(permissions_user),
+            'group': format_permissions(permissions_group),
+            'public': format_permissions(permissions_public),
+        }
+
+    def get_serializer_class(self):
+        # Augment the base serializer class to include permissions information with objects.
+        base_class = super(ResolwePermissionsMixin, self).get_serializer_class()
+
+        class SerializerWithPermissions(base_class):
+            def to_representation(serializer, instance):
+                # TODO: These permissions queries may be expensive. Should we limit or optimize this?
```

Comment:
Thoughts? This could be expensive in large lists. Perhaps we could limit it based on request query parameters.